crates/rpc/src/v03/method/simulate_transaction.rs

```diff
@@ -0,0 +1,500 @@
+    }
+
+    #[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
+    pub struct DeployAccountTxnTrace {
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub constructor_invocation: Option<FunctionInvocation>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub fee_transfer_invocation: Option<FunctionInvocation>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub validate_invocation: Option<FunctionInvocation>,
+    }
+
+    #[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
+    pub struct InvokeTxnTrace {
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub execute_invocation: Option<FunctionInvocation>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub fee_transfer_invocation: Option<FunctionInvocation>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub validate_invocation: Option<FunctionInvocation>,
+    }
+
+    #[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
+    pub struct L1HandlerTxnTrace {
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub function_invocation: Option<FunctionInvocation>,
+    }
+
+    #[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
+    pub struct SimulatedTransaction {
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub fee_estimation: Option<FeeEstimate>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub transaction_trace: Option<TransactionTrace>,
+    }
+
+    #[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
+    pub struct FeeEstimate {
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub gas_consumed: Option<NumAsHex>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub gas_price: Option<NumAsHex>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub overall_fee: Option<NumAsHex>,
+    }
+
+    #[derive(Debug, Deserialize, Serialize, Eq, PartialEq)]
+    pub struct NumAsHex(pub String);
+}
+
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    use pathfinder_common::{felt, Chain};
+    use pathfinder_storage::{JournalMode, Storage};
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_simulate_transaction() {
+        let mut db_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        db_path.push("fixtures/simulate-transaction-test.sqlite");
+
+        if db_path.exists() {
+            std::fs::remove_file(&db_path).expect("remove db");
+        }
+
+        let storage = Storage::migrate(db_path.clone(), JournalMode::WAL).expect("storage");
```

Comment:
Why not use `Storage::in_memory()` instead?

Reply:
I will add clean up though to keep the test hermetic.

Reply:
Hmm, yeah, you're right, I completely forgot about the fact that this has to work through processes...

Yes, some cleanup would be good.

Reply:
Done